KerberosConfigPath cannot be changed without reloading the JDBC driver

The JDBC driver accepts the `KerberosConfigPath` property and validates if it matches the `java.security.krb5.conf` system property, if the latter is set. But then it sets it to this value, which prevents using a different value for a subsequent connection.

I've noticed this when testing Kerberos auth in Tableau Desktop. After spotting a typo in the value, I couldn't correct it, without restarting Tableau. Furthermore, the error visible to the user is `Error setting up connection`, which is not very actionable:

Full stack-trace:
```
2021-08-24 12:00:19.650 +0000 (,,,,5,17) grpc-default-executor-3 : ERROR com.tableau.connect.grpc.GrpcProtocolService - Failed in constructProtocol. SQLState=null, ErrorCode=0.
com.tableausoftware.jdbc.SQLExceptionWithDriverInfo: Error setting up connection
	at com.tableausoftware.jdbc.JDBCDriverManager.getConnection(JDBCDriverManager.java:262) ~[jdbcserver.jar:20212.0.8]
	at com.tableausoftware.jdbc.JDBCProtocolImpl.getConnection(JDBCProtocolImpl.java:320) ~[jdbcserver.jar:20212.0.8]
	at com.tableausoftware.jdbc.JDBCProtocolImpl.createConnectionWithIntegratedAuth(JDBCProtocolImpl.java:264) ~[jdbcserver.jar:20212.0.8]
	at com.tableausoftware.jdbc.JDBCProtocolImpl.<init>(JDBCProtocolImpl.java:115) ~[jdbcserver.jar:20212.0.8]
	at com.tableau.connect.service.ProtocolPool.constructProtocol(ProtocolPool.java:46) ~[jdbcserver.jar:20212.0.8]
	at com.tableau.connect.service.ProtocolService.constructProtocol(ProtocolService.java:58) ~[jdbcserver.jar:20212.0.8]
	at com.tableau.connect.grpc.GrpcProtocolService.lambda$constructProtocol$0(GrpcProtocolService.java:66) ~[jdbcserver.jar:20212.0.8]
	at com.tableau.connect.grpc.GrpcProtocolService.wrap(GrpcProtocolService.java:292) ~[jdbcserver.jar:20212.0.8]
	at com.tableau.connect.grpc.GrpcProtocolService.constructProtocol(GrpcProtocolService.java:65) ~[jdbcserver.jar:20212.0.8]
	at com.tableau.connect.generated.ProtocolServiceGrpc$MethodHandlers.invoke(ProtocolServiceGrpc.java:1492) ~[jdbcserver.jar:20212.0.8]
	at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:180) ~[jdbcserver.jar:20212.0.8]
	at io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35) ~[jdbcserver.jar:20212.0.8]
	at io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23) ~[jdbcserver.jar:20212.0.8]
	at io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:40) ~[jdbcserver.jar:20212.0.8]
	at io.grpc.Contexts$ContextualizedServerCallListener.onHalfClose(Contexts.java:86) ~[jdbcserver.jar:20212.0.8]
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331) ~[jdbcserver.jar:20212.0.8]
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:814) ~[jdbcserver.jar:20212.0.8]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[jdbcserver.jar:20212.0.8]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123) ~[jdbcserver.jar:20212.0.8]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.sql.SQLException: Error setting up connection
	at io.trino.jdbc.TrinoDriverUri.setupClient(TrinoDriverUri.java:332) ~[?:?]
	at io.trino.jdbc.NonRegisteringTrinoDriver.connect(NonRegisteringTrinoDriver.java:56) ~[?:?]
	at com.tableausoftware.jdbc.JDBCDriverManager.getConnection(JDBCDriverManager.java:254) ~[jdbcserver.jar:20212.0.8]
	... 21 more
Caused by: java.lang.IllegalStateException: Refusing to set system property 'java.security.krb5.conf' to 'C:\Windows\krb5.ini', it is already set to 'C:\Windows'\krb5.ini'
	at io.trino.jdbc.$internal.guava.base.Preconditions.checkState(Preconditions.java:826) ~[?:?]
	at io.trino.jdbc.$internal.client.SpnegoHandler.lambda$new$0(SpnegoHandler.java:108) ~[?:?]
	at java.util.Optional.ifPresent(Unknown Source) ~[?:?]
	at io.trino.jdbc.$internal.client.SpnegoHandler.<init>(SpnegoHandler.java:105) ~[?:?]
	at io.trino.jdbc.$internal.client.OkHttpUtil.setupKerberos(OkHttpUtil.java:291) ~[?:?]
	at io.trino.jdbc.TrinoDriverUri.setupClient(TrinoDriverUri.java:289) ~[?:?]
	at io.trino.jdbc.NonRegisteringTrinoDriver.connect(NonRegisteringTrinoDriver.java:56) ~[?:?]
	at com.tableausoftware.jdbc.JDBCDriverManager.getConnection(JDBCDriverManager.java:254) ~[jdbcserver.jar:20212.0.8]
	... 21 more
```

Refs:
https://github.com/trinodb/trino/blob/fe608f2723842037ff620d612a706900e79c52c8/client/trino-client/src/main/java/io/trino/client/SpnegoHandler.java#L109

I suggest removing this check.

--- a/client/trino-client/src/main/java/io/trino/client/auth/kerberos/LoginBasedSubjectProvider.java
+++ b/client/trino-client/src/main/java/io/trino/client/auth/kerberos/LoginBasedSubjectProvider.java
@@ -61,6 +61,11 @@ public class LoginBasedSubjectProvider
                     "Refusing to set system property 'java.security.krb5.conf' to '%s', it is already set to '%s'",
                     newValue,
                     currentValue);
+            checkState(
+                    file.exists() && !file.isDirectory(),
+                    "Kerberos config file '%s' does not exist or is a directory",
+                    newValue);
+            checkState(file.canRead(), "Kerberos config file '%s' is not readable", newValue);
             System.setProperty("java.security.krb5.conf", newValue);
         });
     }

